src/Sulu/Bundle/AdminBundle/Tests/Unit/FormMetadata/StructureFormMetadataLoaderTest.php

```diff
@@ -0,0 +1,163 @@
+<?php
+
+/*
+ * This file is part of Sulu.
+ *
+ * (c) Sulu GmbH
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace Sulu\Bundle\AdminBundle\Tests\Unit\FormMetadata;
+
+use PHPUnit\Framework\TestCase;
+use Prophecy\Prophecy\ObjectProphecy;
+use Sulu\Bundle\AdminBundle\FormMetadata\FormMetadataMapper;
+use Sulu\Bundle\AdminBundle\Metadata\FormMetadata\FieldMetadata;
+use Sulu\Bundle\AdminBundle\Metadata\FormMetadata\FormMetadata;
+use Sulu\Bundle\AdminBundle\Metadata\FormMetadata\ItemMetadata;
+use Sulu\Bundle\AdminBundle\Metadata\FormMetadata\SectionMetadata;
+use Sulu\Bundle\AdminBundle\Metadata\FormMetadata\StructureFormMetadataLoader;
+use Sulu\Bundle\AdminBundle\Metadata\FormMetadata\Validation\FieldMetadataValidatorInterface;
+use Sulu\Bundle\AdminBundle\Metadata\SchemaMetadata\SchemaMetadata;
+use Sulu\Component\Content\Metadata\Factory\StructureMetadataFactoryInterface;
+use Sulu\Component\Content\Metadata\StructureMetadata;
+use Sulu\Component\Webspace\Manager\WebspaceManagerInterface;
+
+class StructureFormMetadataLoaderTest extends TestCase
+{
+    const CACHE_DIR = __DIR__ . '/../../../../../../../tests/Resources/cache';
+
+    /**
+     * @var ObjectProphecy&StructureMetadataFactoryInterface
+     */
+    private $structureMetadataFactory;
+
+    /**
+     * @var ObjectProphecy&FormMetadataMapper
+     */
+    private $formMetadataMapper;
+
+    /**
+     * @var ObjectProphecy&WebspaceManagerInterface
+     */
+    private $webspaceManager;
+
+    /**
+     * @var ObjectProphecy&FieldMetadataValidatorInterface
+     */
+    private $fieldMetadataValidator;
+
+    /**
+     * @var StructureFormMetadataLoader
+     */
+    private $structureFormMetadataLoader;
+
+    protected function setUp(): void
+    {
+        $this->structureMetadataFactory = $this->prophesize(StructureMetadataFactoryInterface::class);
+        $this->formMetadataMapper = $this->prophesize(FormMetadataMapper::class);
+        $this->webspaceManager = $this->prophesize(WebspaceManagerInterface::class);
+        $this->fieldMetadataValidator = $this->prophesize(FieldMetadataValidatorInterface::class);
+
+        $this->structureFormMetadataLoader = new StructureFormMetadataLoader(
+            $this->structureMetadataFactory->reveal(),
+            $this->formMetadataMapper->reveal(),
+            $this->webspaceManager->reveal(),
+            $this->fieldMetadataValidator->reveal(),
+            [],
+            ['en'],
+            static::CACHE_DIR,
+            false
+        );
+    }
+
+    /**
+     * @param FormMetadata[] $types
+     */
+    private function createFieldMetadata(
+        string $name,
+        string $type = 'text_line',
```

Comment:
I think I would even make that one optional... Otherwise it wouldn't be clear if code like that should succeed or fail without looking at the function definition, which might be in a complete different location:

```php
$fieldMetadata = $this->createFieldMetadata('test');
$this->assertEquals('text_line', $fieldMetadata->getType());
```